core/txpool/legacypool/list.go

```diff
@@ -396,6 +396,35 @@ func (l *list) Filter(costLimit *uint256.Int, gasLimit uint64) (types.Transactio
 	return removed, invalids
 }
 
+func (l *list) ClearList() (types.Transactions, types.Transactions) {
+	// Filter out all the transactions
+	removed := l.txs.Filter(func(tx *types.Transaction) bool {
+		return true
+	})
```

Comment:
if this is just removing all txs in `l.txs`, just do `l.txs = newSortedMap()`

Reply:
hmm what i like about `Filter` is that I get back the txs that have been removed. I use this list of removed txs to remove txs from the legacy pools lookup https://github.com/astriaorg/flame/blob/af08e5b2d5c30613609c1db2eaf073fc2456cb6c/core/txpool/legacypool/legacypool.go#L1797, if i had use `newSortedMap` I wouldn't have been able to get this data hmmm